rllib/env/external_env.py

```diff
@@ -378,7 +378,7 @@ def poll(
                 if not self.external_env.is_alive():
                     raise Exception("Serving thread has stopped.")
         limit = self.external_env._max_concurrent_episodes
-        assert len(results[0]) < limit, (
```

Comment:
does this actually solve the problem? Can we add a test from the repro script for this.

Reply:
This is what I'm not sure about: if this actually solves the problem or not. I was working on adding a unit test, but the test is still broken if I play with the value of `rollout_fragment_length`, so I'm confused about the original use case for the `max_concurrent` arg and when it should fail vs when it shouldn't.

Reply:
Putting up a minimal repro example rn

Reply:
Added a unit test: `python -m pytest -rP rllib/env/tests/test_external_env.py -k 'max_concurrent'` fails, basically the `rollout_fragment_length=40` setting causes the env to rollout multiple episodes (25 timesteps each) before it returns control to the algo, which conflicts with the `max_concurrent=1` setting.

My question is basically why do we even want the `max_concurrent` setting in the first place in this way. The way it's currently written, it'll also error out if we use vectorized external envs (again, multiple episodes running concurrently).